### ISSUE TITLE
regexp/optimal-quantifier-concatenation false positive if group is back referenced

### ISSUE THREAD

**Issue by adamlui** [ota-meshi/eslint-plugin-regexp#775]
**Information:**
- ESLint version: 9.14.0
- `eslint-plugin-regexp` version: 2.6.0

**Description**

Take this code:

```js
lang.href = lang.href.replace(/(.*\/\/.*?\/)((\w{2}(-\w{2})?\/)?.*)\.md/, '$1#/$2')
```

Your plugin will error like:

```
error: '(\w{2}(-\w{2})?\/)?' can be removed because it is already included by '.*'. This cannot be fixed automatically because it might change or remove a capturing group  regexp/optimal-quantifier-concatenation
```

However it is false it "can be removed" since the desired `$2` in replacement needs the group, even if already included by `.*`

**Comment by adamlui** [comment]
This is not a bug, I miscounted the capturing group number, your plugin is amazing
